### ISSUE TITLE
`unnecessary_safety_comment` does not lint the first item of a module without a body

### ISSUE THREAD

**Issue by ojeda** [rust-lang/rust-clippy#14554]
### Summary

`unnecessary_safety_comment` does not lint the first item of a module without a body.

Possibly related: https://github.com/rust-lang/rust-clippy/issues/14553.

### Lint Name

`unnecessary_safety_comment`

### Reproducer

I tried this code:

File `a.rs`:

```rust
mod b;
```

File `b.rs`:

```rust
// This line should avoid #14553.

// SAFETY: ...
mod x {}

// SAFETY: ...
mod y {}
```

I expected to see this happen: Two lints, for both safety comments.

Instead, this happened: Only the comment on `y` is linted.

Note that there is no false negative if we embed module `b`:

```rust
mod b {
    // This line should avoid #14553.

    // SAFETY: ...
    mod x {}

    // SAFETY: ...
    mod y {}
}
```

### Version

```text
rustc 1.88.0-nightly (17ffbc81a 2025-04-04)
binary: rustc
commit-hash: 17ffbc81a30c094193836a5d7f90dff273b5df93
commit-date: 2025-04-04
host: x86_64-unknown-linux-gnu
release: 1.88.0-nightly
LLVM version: 20.1.2
```
